src/server/pfs/server/url_worker.go

```diff
@@ -53,23 +55,28 @@ func (d *driver) URLWorker(ctx context.Context) {
 	})
 }
 
-func (d *driver) processPutFileURLTask(ctx context.Context, task *PutFileURLTask) (*types.Any, error) {
+func (d *driver) processPutFileURLTask(ctx context.Context, task *PutFileURLTask) (any *types.Any, retErr error) {
 	url, err := obj.ParseURL(task.URL)
 	if err != nil {
 		return nil, errors.Wrapf(err, "error parsing URL %v", task.URL)
 	}
-	objClient, err := obj.NewClientFromURLAndSecret(ctx, url, false)
+	bucket, err := openBucket(ctx, url)
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
+		if err := bucket.Close(); err != nil {
+			retErr = multierror.Append(retErr, errors.Wrapf(err, "error closing bucket %s", url.Bucket))
+		}
+	}()
 	prefix := strings.TrimPrefix(url.Object, "/")
 	result := &PutFileURLTaskResult{}
 	if err := log.LogStep(ctx, "putFileURLTask", func(ctx context.Context) error {
 		return d.storage.WithRenewer(ctx, defaultTTL, func(ctx context.Context, renewer *fileset.Renewer) error {
 			id, err := d.withUnorderedWriter(ctx, renewer, func(uw *fileset.UnorderedWriter) error {
 				for _, path := range task.Paths {
 					if err := miscutil.WithPipe(func(w io.Writer) error {
```

Comment:
This pipe isn't needed with the new interface. We can simplify this and just pass the bucket reader to the put call.